osu.Game/Screens/Menu/ButtonSystem.cs

```diff
@@ -81,10 +81,66 @@ public void SetOsuLogo(OsuLogo logo)
         private readonly ButtonArea buttonArea;
 
         private readonly MainMenuButton backButton;
+        private readonly MainMenuButton settingsButton;
+
+        // public for tests
+        [CanBeNull]
+        public MainMenuButton PlayButton { get; private set; }
 
         private readonly List<MainMenuButton> buttonsTopLevel = new List<MainMenuButton>();
         private readonly List<MainMenuButton> buttonsPlay = new List<MainMenuButton>();
 
+        [CanBeNull]
+        public List<MainMenuButton> CurrentButtonsList => State switch
+        {
+            ButtonSystemState.TopLevel => buttonsTopLevel,
+            ButtonSystemState.Play => buttonsPlay,
+            _ => null
+        };
+
+        private int selectionIndex = -1;
+
+        public int SelectionIndex
+        {
+            get => selectionIndex;
+            set
+            {
+                MainMenuButton previousButton = null;
+                if (CurrentSelection is MainMenuButton button)
+                    previousButton = button;
+
+                selectionIndex = value;
+
+                if (CurrentSelection is MainMenuButton newButton)
+                {
+                    if (previousButton != newButton)
+                    {
+                        previousButton?.SimulateHoverLost();
+                        newButton.SimulateHover();
+                    }
+                }
+                else
+                    previousButton?.SimulateHoverLost();
```

Comment:
All this is a bit horrific but I don't see a great way of replacing it. That said, you could do

```suggestion
                var previousButton = CurrentSelection as MainMenuButton;
                previousButton?.SimulateHoverLost();

                selectionIndex = value;

                var currentButton = CurrentSelection as MainMenuButton;
                currentButton?.SimulateHover();
```

which doesn't seem to confuse static analysis, I guess.

Reply:
yeah I wrote it this way, so it doesn't call SimulateHoverLost and SimulateHover on the same object, to avoid it confusing the animations. I think this can mostly be skipped by checking for `selectionIndex != value` though